src/conn/conn_prefetch.c

```diff
@@ -106,6 +106,15 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
         locked = false;
 
         WT_WITH_DHANDLE(session, pe->dhandle, ret = __wt_prefetch_page_in(session, pe));
+
```

Comment:
I think this diff was applied incorrectly - as it is, it'll prefetch the page regardless of the corruption flag, then check the corruption flag and prefetch it again. I like the comment this time around though ;-)

By the way - are there a bunch more checks like this going to be added? If so, now might be the time to refactor this into a `__prefetch_can_page_in`, and replace this with (e.g.)

```
if (__prefetch_can_page_in(...))
    WT_WITH_DHANDLE(...);
```

Reply:
Oh good spot on the repeated `prefetch_page_in` calls thanks! I'm happy to leave the check as it is for now, since we currently don't have a bunch more checks of this nature. Will definitely refactor in the future should more checks be added, though.